Sourcery/Parsing/Parser.swift

```diff
@@ -207,28 +230,42 @@ final class Parser {
     ///
     /// - Parameter types: Types and extensions.
     /// - Returns: Just types.
-    internal func uniqueTypes(_ types: [Type]) -> [Type] {
+    internal func uniqueTypes(_ types: [Type], typealiases: [String: String]) -> [Type] {
         var unique = [String: Type]()
+        
+        //replace extensions for type aliases with original types
+        types
+            .filter { $0.isExtension == true }
+            .forEach { $0.localName = typealiases[$0.name] ?? $0.localName }
 
         types
             .filter { $0.isExtension == false }
             .forEach { unique[$0.name] = $0 }
 
         types.forEach { type in
             guard let current = unique[type.name] else {
-                let inheritanceClause = type.inheritedTypes.isEmpty ? "" : ": \(type.inheritedTypes)"
-                if verbose { print("\(logPrefix)Ignoring \"extension \(type.name)\(inheritanceClause)\" because we don't have original type definition information") }
+                unique[type.name] = type
```

Comment:
I don't know to be honest. It all depends on use cases for templates. I'll revert it for now and if there will be such feature request we can put it back.